cosmpy/aerial/contract/__init__.py

```diff
@@ -0,0 +1,250 @@
+
+        # broadcast the store transaction
+        submitted_tx = self._client.broadcast_tx(tx).wait_to_complete()
+
+        # extract the code id
+        self._code_id = submitted_tx.contract_code_id
+        if self._code_id is None:
+            raise RuntimeError(f"Unable to extract contract code id")
+
+        return self._code_id
+
+    def instantiate(
+        self,
+        code_id: int,
+        args: Any,
+        key: PrivateKey,
+        label: Optional[str] = None,
+        gas_limit: Optional[int] = None,
+        admin_address: Optional[Address] = None,
+        funds: Optional[str] = None,
+    ) -> Address:
+        sender_address = Address(key)
+
+        # query the account information for the sender
+        account = self._client.query_account(sender_address)
+        label = label or _generate_label(self._digest)
+
+        # estimate the fee required for this transaction
+        gas_limit = (
+            gas_limit or 2000000
+        )  # TODO: Need to interface to the simulation engine
+        fee = self._client.estimate_fee_from_gas(gas_limit)
+
+        # build up the store transaction
+        tx = Transaction()
+        tx.add_message(
+            create_cosmwasm_instantiate_msg(
+                code_id,
+                args,
+                label,
+                sender_address,
+                admin_address=admin_address,
+                funds=funds,
+            )
+        )
+        tx.seal(SigningCfg.direct(key, account.sequence), fee=fee, gas_limit=gas_limit)
+        tx.sign(key, self._client.network_config.chain_id, account.number)
+        tx.complete()
+
+        # broadcast the store transaction
+        submitted_tx = self._client.broadcast_tx(tx).wait_to_complete()
+
+        # store the contract address
+        self._address = submitted_tx.contract_address
+        if self._address is None:
+            raise RuntimeError(f"Unable to extract contract code id")
+
+        return self._address
+
+    def deploy(
+        self,
+        args: Any,
+        key: PrivateKey,
+        label: Optional[str] = None,
+        store_gas_limit: Optional[int] = None,
+        instantiate_gas_limit: Optional[int] = None,
+        admin_address: Optional[Address] = None,
+        funds: Optional[str] = None,
+    ) -> Address:
+
+        # in the case where the contract is already deployed
+        if self._address is not None and self._code_id is not None:
+            return self._address
+
+        assert self._address is None
+
+        if self._code_id is None:
+            self.store(key, gas_limit=store_gas_limit)
+
+        assert self._code_id is not None
+
+        return self.instantiate(
+            self._code_id,
+            args,
+            key,
+            label=label,
+            gas_limit=instantiate_gas_limit,
+            admin_address=admin_address,
+            funds=funds,
+        )
+
+    def execute(
+        self,
+        args: Any,
+        key: PrivateKey,
```

Comment:
I see you are using PrivateKey instead of CosmosCrypto - I guess we should integrate some functionality from CosmosCrypto to PrivateKey and remove CosmosCrypto

- Or I guess it will be better to pass Crypto here that also works as a PrivateKey. - We should be able to get_address from Crypto for example.